File: src/main/java/com/chattriggers/ctjs/launch/mixins/transformers/ConnectScreenMixin.java
package com.chattriggers.ctjs.launch.mixins.transformers;

import com.chattriggers.ctjs.triggers.TriggerType;
import net.minecraft.client.MinecraftClient;
import net.minecraft.client.gui.screen.ConnectScreen;
import net.minecraft.client.gui.screen.Screen;
import net.minecraft.client.network.ServerAddress;
import net.minecraft.client.network.ServerInfo;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;

@Mixin(ConnectScreen.class)
public class ConnectScreenMixin {
    @Inject(
        method = "connect(Lnet/minecraft/client/gui/screen/Screen;Lnet/minecraft/client/MinecraftClient;Lnet/minecraft/client/network/ServerAddress;Lnet/minecraft/client/network/ServerInfo;)V",
        at = @At("HEAD")
    )
    private static void injectConnect(
        Screen screen,
        MinecraftClient client,
        ServerAddress address,
        ServerInfo info,
        CallbackInfo ci
    ) {
        // TODO(FEATURE): Pass in some args?
        TriggerType.ServerConnect.triggerAll();
    }
}

File: src/main/kotlin/com/chattriggers/ctjs/launch/mixins/transformers/ConnectScreenMixin.kt
package com.chattriggers.ctjs.launch.mixins.transformers

import com.chattriggers.ctjs.triggers.TriggerType
import net.minecraft.client.MinecraftClient
import net.minecraft.client.gui.screen.ConnectScreen
import net.minecraft.client.gui.screen.Screen
import net.minecraft.client.network.ServerAddress
import net.minecraft.client.network.ServerInfo
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.injection.At
import org.spongepowered.asm.mixin.injection.Desc
import org.spongepowered.asm.mixin.injection.Inject

@Mixin(ConnectScreen::class)
class ConnectScreenMixin {
    @Inject(
        target = [Desc(
            value = "connect",
            args = [
                Screen::class, MinecraftClient::class, ServerAddress::class, ServerInfo::class
            ],
        )],
        at = [At("HEAD")],
    )
    fun injectConnect() {
        // TODO(FEATURE): Pass in some args?
        TriggerType.ServerConnect.triggerAll()
    }
}
